File: LinkedList/15/sol.py
# A linked list (LL) node 
# to store a queue entry 
class Node: 
      
    def __init__(self, data): 
        self.data = data 
        self.next = None
        
class MyQueue:
    
    
     # Method to add an item to the queue
     def __init__(self):
         self.head = None
     def push(self, item): 
         if self.head is None:
             self.head = Node(item)
             return
         ptr = self.head
         while ptr.next is not None:
             ptr = ptr.next
         ptr.next = Node(item)
         return
         #Add code here
    
    # Method to remove an item from queue
     def pop(self):
         if self.head is None:
             return -1
         data = self.head.data
         self.head = self.head.next
         return data
         
         #add code here



#{ 
#  Driver Code Starts
if __name__=='__main__':
    t=int(input())
    for i in range(t):
        s=MyQueue()
        q=int(input())
        q1=list(map(int,input().split()))
        i=0
        while(i<len(q1)):
            if(q1[i]==1):
                s.push(q1[i+1])
                i=i+2
            elif(q1[i]==2):
                print(s.pop(),end=" ")
                i=i+1
            elif(s.isEmpty()):
                print(-1)
                i=i+1
        print()   
# } Driver Code Ends

File: LinkedList/12/sol.java
// { Driver Code Starts
import java.util.*;
import java.io.*;

class Node{
    int data;
    Node next;
    
    Node(int x){
        data = x;
        next = null;
    }
    
}
class GFG{
	static void printList(Node node) 
	{ 
		while (node != null) 
		{ 
			System.out.print(node.data + " "); 
			node = node.next; 
		} 
		System.out.println(); 
	}
	
    public static void main(String args[]) throws IOException { 
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t > 0){
        	int n = sc.nextInt();
            
            Node head = new Node(sc.nextInt());
            Node tail = head;
            for(int i=0; i<n-1; i++)
            {
                tail.next = new Node(sc.nextInt());
                tail = tail.next;
            }
            
            Solution g = new Solution();
            head = g.pairwiseSwap(head);
            printList(head); 
            t--;
        }
    } 
} 
   
// } Driver Code Ends




/* node class of the linked list

class Node
{
    int data;
    Node next;
    Node(int key)
    {
        data = key;
        next = null;
    }
}

*/


class Solution {
    // Function to pairwise swap elements of a linked list.
    // It should returns head of the modified list
    public Node pairwiseSwap(Node head)
    {
        Node one,two,ptr,myHead = null;
        ptr = head;
        boolean flag = true;
        while (ptr != null)
        {
            one = ptr;
            if(one.next == null)break;
            two = one.next;
            if(flag)
            {
                head = two;
                flag = false;
            }
            else
            {
                myHead.next = two;
            }
            ptr = two.next;
            one.next = two.next;
            two.next = one;
            myHead = one;
        }
        return head;
        // code here
    }
}